tools/topology/topology2/include/pipelines/cavs/gain-copier-capture.conf

```diff
@@ -45,59 +46,89 @@ Class.Pipeline."gain-copier-capture" {
 	Object.Widget {
 		host-copier."1" {
 			type	"aif_out"
-			num_input_audio_formats 3
+			num_input_audio_formats 1
 			num_output_audio_formats 3
 			num_input_pins 1
 
-			# 16-bit 48KHz 2ch
-			Object.Base.audio_format.1 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-			}
-			# 24-bit 48KHz 2ch
-			Object.Base.audio_format.2 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	24
-			}
-			# 32-bit 48KHz 2ch
-			Object.Base.audio_format.3 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	32
-			}
+			Object.Base.input_audio_format [
+				# 32-bit 48KHz 2ch
+				{
+					in_bit_depth		32
+					in_valid_bit_depth	32
+				}
+			]
+
+			Object.Base.output_audio_format [
+				{
+					out_bit_depth		32
+					out_valid_bit_depth	32
+				}
+				{
+					out_bit_depth		16
+					out_valid_bit_depth	16
+				}
+				{
+					out_bit_depth		32
+					out_valid_bit_depth	24
+				}
+			]
 			node_type $HDA_HOST_INPUT_CLASS
 		}
 
 		gain."1" {
 			num_input_audio_formats 2
 			num_output_audio_formats 2
-			#16-bit 48KHz 2ch
-			Object.Base.audio_format.1 {}
-			# 32-bit 48KHz 2ch
-			Object.Base.audio_format.2 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	32
-			}
+			Object.Base.input_audio_format [
+				#16-bit 48KHz 2ch
+				{
+					in_bit_depth		16
+					in_valid_bit_depth	16
+				}
+				# 32-bit 48KHz 2ch
+				{
+					in_bit_depth		32
+					in_valid_bit_depth	32
+				}
+			]
+
+			Object.Base.output_audio_format [
+				{
+					out_bit_depth		32
+					out_valid_bit_depth	32
+				}
+				{
+					out_bit_depth		16
+					out_valid_bit_depth	16
+				}
```

Comment:
Does the order matter? The existing code format 1 is 16 bit and format 2 is 32 bit. But after the change output format 1 is 32 bit and output format 2 is 16 bit.

Reply:
order in topology should not matter as FW should iterate over configs to match requested based on copier use case.

Reply:
yes, order doesnt matter